seis/ms/header.go

```diff
@@ -0,0 +1,359 @@
+const RecordHeaderSize = 48
+
+type RecordHeader struct {
+	SequenceNumber [6]byte // ASCII String representing a 6 digit number
+
+	DataQualityIndicator byte // ASCII: D, R, Q or M
+	ReservedByte         byte // ASCII: Space
+
+	// These are ascii strings
+	StationIdentifier  [5]byte // ASCII: Left justify and pad with spaces
+	LocationIdentifier [2]byte // ASCII: Left justify and pad with spaces
+	ChannelIdentifier  [3]byte // ASCII: Left justify and pad with spaces
+	NetworkIdentifier  [2]byte // ASCII: Left justify and pad with spaces
+
+	RecordStartTime      BTime  // Start time of record
+	NumberOfSamples      uint16 // Number of Samples in the data block which may or may not be unpacked.
+	SampleRateFactor     int16  // >0: Samples/Second <0: Second/Samples 0: Seconds/Sample, ASCII/OPAQUE DATA records
+	SampleRateMultiplier int16  // >0: Multiplication Factor <0: Division Factor
+
+	// Flags are bit masks
+	ActivityFlags    byte
+	IOAndClockFlags  byte
+	DataQualityFlags byte
+
+	NumberOfBlockettesThatFollow uint8 // Total number of blockettes that follow
+
+	TimeCorrection  int32  // 0.0001 second units
+	BeginningOfData uint16 // Offset in bytes to the beginning of data.
+	FirstBlockette  uint16 // Offset in bytes to the first data blockette in the data record.
+}
+
+func (h RecordHeader) SeqNumber() int {
+	s := string(h.SequenceNumber[:])
+	if no, err := strconv.ParseInt(s, 10, 32); err == nil {
+		return int(no)
+	}
+	return 0
+}
+func (h *RecordHeader) SetSeqNumber(no int) {
+	copy(h.SequenceNumber[:], []byte(fmt.Sprintf("%06d", no)))
+}
+
+func (h RecordHeader) Station() string {
+	return strings.TrimSpace(string(h.StationIdentifier[:]))
+}
+func (h *RecordHeader) SetStation(s string) {
+	for len(s) < 5 {
+		s += " "
+	}
+	copy(h.StationIdentifier[:], s)
+}
+func (h RecordHeader) Location() string {
+	return strings.TrimSpace(string(h.LocationIdentifier[:]))
+}
+func (h *RecordHeader) SetLocation(s string) {
+	for len(s) < 2 {
+		s += " "
+	}
+	copy(h.LocationIdentifier[:], s)
+}
+func (h RecordHeader) Channel() string {
+	return strings.TrimSpace(string(h.ChannelIdentifier[:]))
+}
+func (h *RecordHeader) SetChannel(s string) {
+	for len(s) < 3 {
+		s += " "
+	}
+	copy(h.ChannelIdentifier[:], s)
+}
+func (h RecordHeader) Network() string {
+	return strings.TrimSpace(string(h.NetworkIdentifier[:]))
+}
+func (h *RecordHeader) SetNetwork(s string) {
+	for len(s) < 2 {
+		s += " "
+	}
+	copy(h.NetworkIdentifier[:], s)
+}
+func (h RecordHeader) SrcName(quality bool) string {
+	if quality {
+		return strings.Join([]string{h.Network(), h.Station(), h.Location(), h.Channel(), string(h.DataQualityIndicator)}, "_")
+	}
+	return strings.Join([]string{h.Network(), h.Station(), h.Location(), h.Channel()}, "_")
+}
+
+func (h *RecordHeader) SetStartTime(t time.Time) {
+	h.RecordStartTime = NewBTime(t)
+}
+
+func (h *RecordHeader) SetCorrection(correction time.Duration, applied bool) {
+	switch applied {
```

Comment:
would use `if` statement in stead

Reply:
this is a style thing, I prefer the switch mechanism rather than a stack of if elses. but can change if it's hard to read.

https://golang.org/doc/effective_go#switch

> It's therefore possible—and idiomatic—to write an if-else-if-else chain as a switch.

although a single else may not exactly be a chain, it's an idiomatic convention I've fallen into.